gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/sql/execute/AbstractGemFireDistributionActivation.java

```diff
@@ -640,6 +644,7 @@ protected final Object operateOnNonPartitionedRegion(
           (isForSelectForUpdateQuery && this.qInfo.needKeysForSelectForUpdate()),
           this.pvs, rgn, null, this.qInfo.isInsertAsSubSelect(), lcc, this
               .getTimeOutMillis());
+      functionMsg.setAbortOnLowMemory(GfxdHeapThresholdListener.isCancellableQuery(this));
       setAllAreNonPartitionedRegion(rs, isVTIInvolved);
```

Comment:
Will it not be better to set in the constructor of StatementExecutorMessage/PrepStatementExecutorMessage so that it never gets missed by callers (e.g. if someone adds another call to those messages elsewhere later it will be easy to miss this with above).